### PR TITLE
Transferred stuff from old CHAOS-42 branch here

### PR DESCRIPTION
**Pull request by evan-a-w**
Made a new branch for CHAOS-42 since the old branch is super outdated and merging would be a massive effort. I know this isn't best practices but there isn't that much changed so its surely fine :))

### CODE REVIEW

File: backend/server/src/role.rs [hunk 1]
```diff
@@ -105,3 +108,42 @@ pub async fn new_role(
...
+    // status of the campaign.
+
+    db.run(move |conn| {
+        let role = Role::get_from_id(conn, role_id).ok_or(Json(QuestionsError::RoleNotFound))?;
+        let campaign = Campaign::get_from_id(conn, role.campaign_id)
+            .ok_or(Json(QuestionsError::CampaignNotFound))?;
+
+        // Prevent people from viewing while it's in draft mode,
+        // unless they have adequate permissions
+        OrganisationUser::campaign_admin_level(campaign.id, user.id, &conn)
```
Comment (josephjeong) [inline backend/server/src/role.rs:136]: role_admin_level() exists in organisationUser - could just use that

**Review by josephjeong — CHANGES_REQUESTED** [review]

File: backend/server/src/role.rs (hunk 1 above)
Comment (evan-a-w) [reply backend/server/src/role.rs:136]: True, but we need campaign anyway to use campaign.published in the check.

Comment (josephjeong) [reply backend/server/src/role.rs:136]: fair enough, lgtm otherwise :)

**Review by josephjeong — APPROVED** [review]
lgtm!

**Pull request merged**